workflow/controller/operator.go

```diff
@@ -1338,6 +1338,16 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 		if err != nil {
 			return woc.markNodeError(retryNodeName, err), err
 		}
+		if lastChildNode != nil && lastChildNode.Pending() && processedTmpl.GetType() == wfv1.TemplateTypeContainer {
+			_, err := woc.createWorkflowPod(lastChildNode.Name, *processedTmpl.Container, processedTmpl, false)
+			if apierr.IsForbidden(err) {
+				return woc.markNodePending(retryParentNode.Name, err), nil
```

Comment:
Also, just noticed this again. The correct code here should be:

- Mark the child node with the desired phase and message
- Return the parent node

Before you would mark and return the child node and now you mark and return the parent node, both of which are not desired. Sorry if my last comment wasn't clear.

Reply:
Something like:

```go
				_ = woc.markNodePending(lastChildNode.Name, err)
                                return retryParentNode, nil
```